pennylane/ops/op_math/linear_combination.py

```diff
@@ -0,0 +1,482 @@
+                    str(param) for param in ob.parameters
+                )  # Converts params into immutable type
+                if isinstance(ob, qml.GellMann):
+                    parameters += (ob.hyperparameters["index"],)
+                tensor.append((ob.name, ob.wires, parameters))
+            data.add((co, frozenset(tensor)))
+
+        return data
+
+    def compare(self, other):
+        r"""Determines whether the operator is equivalent to another.
+
+        Currently only supported for :class:`~LinearCombination`, :class:`~.Observable`, or :class:`~.Tensor`.
+        LinearCombinations/observables are equivalent if they represent the same operator
+        (their matrix representations are equal), and they are defined on the same wires.
+
+        .. Warning::
+
+            The compare method does **not** check if the matrix representation
+            of a :class:`~.Hermitian` observable is equal to an equivalent
+            observable expressed in terms of Pauli matrices, or as a
+            linear combination of Hermitians.
+            To do so would require the matrix form of LinearCombinations and Tensors
+            be calculated, which would drastically increase runtime.
+
+        Returns:
+            (bool): True if equivalent.
+
+        **Examples**
+
+        >>> H = qml.LinearCombination(
+        ...     [0.5, 0.5],
+        ...     [qml.PauliZ(0) @ qml.PauliY(1), qml.PauliY(1) @ qml.PauliZ(0) @ qml.Identity("a")]
+        ... )
+        >>> obs = qml.PauliZ(0) @ qml.PauliY(1)
+        >>> print(H.compare(obs))
+        True
+
+        >>> H1 = qml.LinearCombination([1, 1], [qml.PauliX(0), qml.PauliZ(1)])
+        >>> H2 = qml.LinearCombination([1, 1], [qml.PauliZ(0), qml.PauliX(1)])
+        >>> H1.compare(H2)
+        False
+
+        >>> ob1 = qml.LinearCombination([1], [qml.PauliX(0)])
+        >>> ob2 = qml.Hermitian(np.array([[0, 1], [1, 0]]), 0)
+        >>> ob1.compare(ob2)
+        False
+        """
+
+        if (pr1 := self.pauli_rep) is not None and (pr2 := other.pauli_rep) is not None:
+            pr1.simplify()
+            pr2.simplify()
+            return pr1 == pr2
+
+        if isinstance(other, (LinearCombination, qml.Hamiltonian)):
+            op1 = self.simplify()
+            op2 = other.simplify()
+            return op1._obs_data() == op2._obs_data()  # pylint: disable=protected-access
+
+        if isinstance(other, (Tensor, Observable)):
+            op1 = self.simplify()
+            return op1._obs_data() == {
+                (1, frozenset(other._obs_data()))  # pylint: disable=protected-access
+            }
+
+        raise ValueError(
+            "Can only compare a LinearCombination, and a LinearCombination/Observable/Tensor."
+        )
+
+    def __matmul__(self, other):
+        """The product operation between Operator objects."""
+        if isinstance(other, LinearCombination):
+            coeffs1 = copy(self.coeffs)
+            ops1 = self.ops.copy()
```

Comment:
Why the need for copies?

Reply:
Honestly, this has been copy-pasted from the Hamiltonian class, I can try without